### PR TITLE
Redesign Navigation: Dynamic, shared data source.

### PR DESCRIPTION
**Pull request by kylorhall**
See #248, this is an exploratory deviation from that—basically migrating spaghetti code into a bit less spaghetti…maybe it's just a different pasta.  Ruby isn't built for this.

---

This allows us to use the same menu items in both Mobile and Desktop as otherwise I think I'll just have to have `icon: '…', title:'…'` duplicated for every link in a lot of places..  @irwalker Is this a good approach..?  Do you think you could add/edit/modify a link in the future?  I would, of course, have test coverage that asserts 1:1 all of this as well…

I could probably even migrate this to a JSON format, if that helps…eg. use `title_from_page: 'pricing'` to execute on the fly, instead of `locale_page(page: 'pricing', locale_obj: locale_obj)[:page_title]` in instantiation…I think that maybe adds _more complexity_, not less.

Ideally I'd move this into another file, but I can't see how, eg. there are no controllers here…. I could move this into a helper, eg. `helpers/navigation_helper.rb` and have a method like `self.menu_definition` which returns this Array/Data Object.  Then I'd call it like `menu = NavigationHelper.menu_definition()`...  Is that a good idea?  Maybe that's a slightly easier way to test—I could assert like everything has a `columns` or whatnot, alongside some UI tests?  Basically, assert it matches an expected interface..

If this is the way, is there a `jsdoc` like way to document this?  This is a complex data structure; I'd like to define what a `menu` and a `row` and a `column` and a `link` could be…though maybe my test coverage above could really do that (but even then, it'd still be messy).

🤮 I hate Ruby syntax here…and this repo is so far off from being able to look at Ruby Components or whatever they call them, so I can't think of a much better way.

🤯

### CODE REVIEW

File: source/partials/header/_desktop_navigation.html.erb [hunk 1]
```diff
@@ -1,259 +1,39 @@
...
-
-        <% data.blog.posts.values.first(3).each do |blog_post| %>
-          <%= partial "partials/header/link", locals: {
-            title: blog_post[:title],
-            href: post_url(blog_post),
-            truncate: 75,
-            compact: true
-          } %>
+<ul class="nav__menubar" role="menubar" aria-label="Sharesight Navigation Menu">
+  <% menus.each_with_index do |menu, index| %>
```
Comment (irwalker) [inline source/partials/header/_desktop_navigation.html.erb:6]: We could always move this to a helper as well

**Review by irwalker — COMMENTED** [review]
Yeah actually this approach is much easier to follow. Moving the menus stuff into a helper would be my approach. I'd move more to helpers if possible to get some of the more complicated and blargh `.erb` stuff out of there.

File: source/partials/header/_desktop_navigation.html.erb (hunk 1 above)
Comment (kylorhall) [reply source/partials/header/_desktop_navigation.html.erb:6]: Not sure what you mean here…I think in the `-mobile` version, I split this into `_row` and `_column`.  Let me merge these together and re-request a fuller review (minus full tests) that I'll put into _testing_.

**Comment by kylorhall** [comment]
Into #248 now.

**Pull request merged**